routes/api/post.js

```diff
@@ -56,7 +56,13 @@ router.route('/byid/:id')
         try {
             let post = await db.Post.findOne({ "_id": req.params.id }).lean();
             let comments = await db.Comment.find({ postItem: req.params.id }).populate("author").lean();
-            res.status(200).json({ ...post, comments });
+            let likes = await db.Likes.aggregate([
+                { $match: { postItem: mongoose.Types.ObjectId(req.params.id) } },
+                { $group: { _id: '$type', count: { $sum: 1 } } }
+            ]);
+            let likesObj = {};
+            likes.forEach(el => { likesObj[el._id] = el.count });
```

Comment:
this is nice use case for reduce 😊
```suggestion
            const likesById = likes.reduce((likesById, like) => ({...likesById, [like._id]: like.count }), {});
```